private/pkg/netext/netext.go

```diff
@@ -37,8 +38,12 @@ func ValidateHostname(hostname string) (string, error) {
 	}
 
 	parsedHost := hostname
-	if host, _, err := net.SplitHostPort(hostname); err == nil {
-		parsedHost = host
+	if host, port, err := net.SplitHostPort(hostname); err == nil {
+		// net.SplitHostPort performs very lax validation of ports (allowing to be resolved from /etc/services).
+		// Only accept numeric ports here.
+		if portNum, err := strconv.ParseUint(port, 10, 32); err == nil && portNum <= 65535 {
```

Comment:
There were two options here:
- Use strconv and validation (this matches protovalidate-go).
- Allow non-numeric ports and use `net.LookupPort("ip", port)` here - https://pkg.go.dev/net#LookupPort. This requires calling out to the name resolver and felt like overkill here.

Reply:
Ref: https://github.com/bufbuild/protovalidate-go/blob/ec23f34e22ff982547206a8f7c86bc0bf2ad4091/celext/lib.go#L537-L540